v3.0/reference/best-practices/massive-regions.md

```diff
@@ -0,0 +1,149 @@
+
+## Performance problem
+
+From the Raftstore workflow diagram, messages in each Region are processed one by one. When a large number of Regions exist, it takes Raftstore some time to process the heartbeats of these Regions, which can cause some delay. As a result, some read and write requests are not processed in time. If read and write pressure is high, the CPU usage of the Raftstore thread might easily become the bottleneck, which further increases the delay and affects the performance.
+
+Generally, if the CPU usage of the loaded Raftstore reaches 85% or higher, Raftstore goes into a busy state and becomes the bottleneck. At the same time, `propose wait duration` can be as high as hundreds of milliseconds.
+
+> **Note:**
+>
+> + For the CPU usage of Raftstore as mentioned above, Raftstore is single-threaded. If Raftstore is multi-threaded, you can increase the CPU usage threshold (85%) proportionally.
+> + Because I/O operations exist in the Raftstore thread, CPU usage cannot reach 100%.
+
+### Performance monitoring
+
+You can check the following monitoring metrics in Grafana's **TiKV Dashboard**:
+
++ `Raft store CPU` in the **Thread-CPU** panel
+
+    Reference value: lower than `raftstore.store-pool-size * 85%`. TiDB v2.1 does not have the `raftstore.store-pool-size`  configuration item, so you can take this item's value as `1` in v2.1 versions.
+
+    ![Check Raftstore CPU](/media/best-practices/raft-store-cpu.png)
+
++ `Propose wait duration` in the **Raft Propose** panel
+
+    `Propose wait duration` is the delay between the time a request is sent to Raftstore and the time Raftstore actually starts processing the request. Long delay means that Raftstore is busy, or that processing the append log is time-consuming, making Raftstore unable to process the request in time.
+
+    Reference value: lower than 50~100 ms according to the cluster size
+
+    ![Check Propose wait duration](/media/best-practices/propose-wait-duration.png)
+
+## Performance tuning methods
+
+After finding out the cause of a performance problem, try to solve it from the following two aspects:
+
++ Reduce the number of Regions on a single TiKV instance
++ Reduce the number of messages for a single Region
+
+### Method 1: Increase the number of TiKV instances
```

Comment:
Can you move method 1 and 2 back to method 3, 4, 5

Reply:
Is it only for v3.0? What about dev and v3.1?

Reply:
same for dev and v3.1

Reply:
OK